tests/services/test_jwt_service.py

```diff
@@ -0,0 +1,34 @@
+import unittest
+from unittest.mock import patch, MagicMock
+from app.core.exceptions.token import DecodeTokenException
+from app.services.jwt_service import JwtService
+
+class TestJwtService(unittest.IsolatedAsyncioTestCase):
+    async def asyncSetUp(self):
+        self.jwt_service = JwtService()
+
+    @patch('app.services.jwt_service.TokenHelper')
+    async def test_verify_token_valid(self, mock_token_helper):
+        mock_token_helper.decode.return_value = {"sub": "refresh", "id": 1, "onyen": "test_user"}
+        await self.jwt_service.verify_token("valid_token")
+        mock_token_helper.decode.assert_called_once_with(token="valid_token")
+
+    @patch('app.services.jwt_service.TokenHelper')
+    async def test_verify_token_invalid(self, mock_token_helper):
+        mock_token_helper.decode.side_effect = DecodeTokenException()
+        with self.assertRaises(DecodeTokenException):
+            await self.jwt_service.verify_token("invalid_token")
+
+    @patch('app.services.jwt_service.TokenHelper')
+    async def test_refresh_access_token(self, mock_token_helper):
+        mock_token_helper.decode.return_value = {"sub": "refresh", "id": 1, "onyen": "test_user"}
+        mock_token_helper.encode.return_value = "new_access_token"
+
+        result = await self.jwt_service.refresh_access_token("refresh_token")
+
+        mock_token_helper.decode.assert_called_once_with(token="refresh_token")
```

Comment:
I don't think this `assert_called_once_with()` statement is doing much. I think the next one is useful because you're testing that the method gets called with a specific payload which is modified from the input, but this is just checking that `decode()` is called with `refresh_token`, which I don't think is that useful. Plus, we might want to one day change the way the `decode()` method works, and we don't want the tests to fail just because we changed the black box inside `refresh_access_token()` imo.

Reply:
The refresh token would not be accurate unless we ran the decode method on it. It's a simple test, but it serves the function of making sure we are decoding the token before we validate & throw the exception. I agree with your previous comment however that the last unit test already covers the first one. So I can remove `test_verify_token_valid` given it's redundant. The decode method is coming from the utility so I doubt that would impact our code unless they deprecate, which is a separate issue

Reply:
Ah okay, I agree, let's keep this one as is